UI/ChatUI/ChatUI/ChatUI/MainPage.xaml

```diff
@@ -0,0 +1,427 @@
+﻿<Page
+	x:Class="ChatUI.MainPage"
+	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	xmlns:android="http://uno.ui/android"
+	xmlns:behaviors="using:ChatUI.Behaviors"
+	xmlns:converters="using:ChatUI.Converters"
+	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	xmlns:ios="http://uno.ui/ios"
+	xmlns:local="using:ChatUI"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	xmlns:xamarin="http://uno.ui/xamarin"
+	Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
+	mc:Ignorable="d ios android xamarin">
+
+	<Page.Resources>
+		<ResourceDictionary>
+			<converters:FromNullableBoolToVisibilityConverter x:Key="BooleanToVisibility" VisibilityIfTrue="Visible" />
+
+			<Style TargetType="ListView">
+				<Setter Property="RenderTransformOrigin" Value=".5,.5" />
```

Comment:
This template also does not support mouse wheel properly, you have to add:

```suggestion
				<Setter Property="(local:ReversedPointerWheel.IsEnabled)" Value="True" />
				<Setter Property="RenderTransformOrigin" Value=".5,.5" />
```

and also add in your project (or the toolkit)

```csharp
	public static class PointerWheel
	{
		public static readonly DependencyProperty IsReversedProperty = DependencyProperty.RegisterAttached(
			"IsReversed",
			typeof(bool),
			typeof(FrameworkElement),
			new PropertyMetadata(default(bool), OnIsReversedChanged));

		public static bool GetIsReversed(FrameworkElement element)
			=> (bool)element.GetValue(IsReversedProperty);

		public static void SetIsReversed(FrameworkElement element, bool value)
			=> element.SetValue(IsReversedProperty, value);

		private static void OnIsReversedChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
		{
			if (d is ScrollContentPresenter scp)
			{
#if HAS_UNO
				// We do have native support for reversed PointerWheel in Uno.
				Uno.UI.Xaml.Controls.ScrollContentPresenter.SetIsPointerWheelReversed(scp, (bool)e.NewValue);
#else
				scp.PointerWheelChanged -= OnWheelChanged;
				if (e.NewValue is true)
				{
					scp.PointerWheelChanged += OnWheelChanged;
				}
#endif
			}
			else if (d is FrameworkElement sv)
			{
				sv.Loaded -= PropagateValueToScp;
				if (e.NewValue != DependencyProperty.UnsetValue)
				{
					sv.Loaded += PropagateValueToScp;
					PropagateValueToScp(sv, null);
				}

				static void PropagateValueToScp(object sender, RoutedEventArgs? _)
				{
					if (sender is FrameworkElement sv && TryFindFirstChild(sv, 10, out ScrollContentPresenter scp))
					{
						SetIsReversed(scp, GetIsReversed(sv));
					}
				}
			}
		}

		private static void OnWheelChanged(object sender, PointerRoutedEventArgs e)
		{
			if (sender is not ScrollContentPresenter { ScrollOwner: ScrollViewer sv } scp || !GetIsReversed(scp))
			{
				if (sender is FrameworkElement fe)
				{
					fe.PointerWheelChanged -= OnWheelChanged;
				}

				return;
			}

			var properties = e.GetCurrentPoint(null).Properties;
			if (e.KeyModifiers == VirtualKeyModifiers.Control)
			{
				// Zoom, do nothing.
			}
			else if (!scp.CanVerticallyScroll || properties.IsHorizontalMouseWheel || e.KeyModifiers == VirtualKeyModifiers.Shift)
			{
				if (scp.CanHorizontallyScroll)
				{
					sv.ChangeView(
						horizontalOffset: sv.HorizontalOffset + properties.MouseWheelDelta,
						verticalOffset: null,
						zoomFactor: null,
						disableAnimation: false);

					e.Handled = true;
				}
			}
			else
			{
				sv.ChangeView(
					horizontalOffset: null,
					verticalOffset: sv.VerticalOffset + properties.MouseWheelDelta,
					zoomFactor: null,
					disableAnimation: false);

				e.Handled = true;
			}
		}

		private static bool TryFindFirstChild<T>(DependencyObject element, uint limit, /*[NotNullWhen(true)]*/ out T result)
		{
			// Finds the first child of type T in the visual tree of element.
			// This is a workaround for the fact that ScrollViewer doesn't expose its ScrollContentPresenter.

			if (element is T t)
			{
				result = t;
				return true;
			}

			if (limit is not 0)
			{
				for (var i = 0; i < VisualTreeHelper.GetChildrenCount(element); i++)
				{
					if (TryFindFirstChild(VisualTreeHelper.GetChild(element, i), limit - 1, out result))
					{
						return true;
					}
				}
			}

			result = default;
			return false;
		}
	}
```

Reply:
Thank you @dr1rrb. Regarding your `PointerWheel` class, is the `Uno.UI.Xaml.Controls.ScrollContentPresenter.SetIsPointerWheelReversed` type supposed to be referenceable from `Uno.WinUI`? This is a .NET 7 / WinUI Uno project.

Reply:
Yeah `Uno.WinUI`. You will need a fairly recent version of uno to have that API.

Reply:
I have added this in the form of an attached behavior.

Reply:
You forgot to enable the behavior in the style :)

Reply:
I didn't forget, I actually just attached it to the `ListView` itself because I wasn't able to make it work via the styles, but I've made some fixes now and it should be ok.